src/utils/stream.ts

```diff
@@ -0,0 +1,43 @@
+export class StreamingApi {
+  private writer: WritableStreamDefaultWriter<Uint8Array>
+  private encoder: TextEncoder
+  private writable: WritableStream
+
+  constructor(writable: WritableStream) {
+    this.writable = writable
+    this.writer = writable.getWriter()
+    this.encoder = new TextEncoder()
+  }
+
+  async write(input: Uint8Array | string) {
+    if (typeof input === 'string') {
+      input = this.encoder.encode(input)
+    }
+    await this.writer.write(input)
+    return this
+  }
+
+  async writeln(input: string) {
+    await this.write(input + '\n')
+    return this
+  }
+
+  async log(arg: any) {
+    await this.writeln(JSON.stringify(arg))
+    return this
+  }
+
+  wait(ms: number) {
```

Comment:
On second thought about this, I think the name `sleep()` would be better. What do you think?

Reply:
That is more intuitive. Let's do that.